app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class AddressesController < ApplicationController
+  def update
+    service = Service.find(params[:service_id])
+    address = Address.find(params[:id])
```

Comment:
Since these will raise an exception if either the Service or the Address don't exist, you should try to gracefully handle those cases instead of letting it cause an unhandled exception on the server side. Letting unhandled exceptions bubble up will result in a 500 error, which indicates that something unexpected happened on the server side. Not finding a Service or an Address is not an unexpected scenario, though, so instead we should handle that case and return some kind of 4xx-level error to indicate that the client passed invalid IDs.

See https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find for the documentation that describes the error-raising behavior.

Reply:
@richardxia Great advice! I'll just use the find_by_id methods here to not throw errors and just make sure both are truthy (otherwise rendering a bad_request).